perceval/backends/core/pagure.py

```diff
@@ -0,0 +1,528 @@
+        """Fetch the issues from the repository.
+
+        The method retrieves, from a Pagure repository, the issues
+        updated since the given date.
+
+        :param from_date: obtain issues updated since this date
+
+        :returns: a generator of issues
+        """
+        payload = {
+            'status': 'all',
+            'per_page': self.max_items,
+            'order': 'asc',
+        }
+
+        if from_date:
+            payload['since'] = from_date
+
+        path = urijoin("issues")
+        return self.fetch_items(path, payload)
+
+    def user(self, name):
+        """Get the user information and update the user cache
+
+        :param name: username of the user
+
+        :returns: a user object
+        """
+        user = None
+
+        if name in self._users:
+            return self._users[name]
+
+        url_user = urijoin(self.base_url, 'user', name)
+
+        logger.debug("Getting info for %s" % url_user)
+
+        r = self.fetch(url_user)
+        user = r.text
+        self._users[name] = user
+
+        return user
+
+    def fetch(self, url, payload=None, headers=None, method=HttpClient.GET, stream=False, auth=True):
+        """Fetch the data from a given URL.
+
+        :param url: link to the resource
+        :param payload: payload of the request
+        :param headers: headers of the request
+        :param method: type of request call (GET or POST)
+        :param stream: defer downloading the response body until the response content is available
+        :param auth: auth of the request
+
+        :returns a response object
+        """
+        if not self.from_archive:
+            self.sleep_for_rate_limit()
+
+        # In case the issue tracker is disabled for a repository
+        # an HTTP 404 response is returned
+        try:
+            response = super().fetch(url, payload, headers, method, stream, auth)
+        except requests.exceptions.HTTPError:
+            print("The issue tracker is disabled please enable the feature for the repository")
+            exit(1)
+            return
+
+        if not self.from_archive:
+            self.update_rate_limit(response)
+
+        return response
+
+    def fetch_items(self, path, payload):
+        """Return the items from Pagure API using links pagination
+
+        :param path: Path from which the item is to be fetched
+        :param payload: Payload to be added to the request
+
+        :returns: an item object
+        """
+
+        page = 0  # current page
+        last_page = None  # last page
+        if self.namespace:  # if project is under a namspace
+            url_next = urijoin(self.base_url, self.namespace, self.repository, path)
+        else:  # if project is created without a namespace
+            url_next = urijoin(self.base_url, self.repository, path)
+        logger.debug("Get Pagure paginated items from " + url_next)
+
+        response = self.fetch(url_next, payload=payload)
```

Comment:
after this line, the following code should be added:
```
        if not response:
            return []
```

This change is needed due to the comment at https://github.com/chaoss/grimoirelab-perceval/pull/621/files#diff-23aec446f77422330c2dcce453fbc909R423